src/NHibernate/Linq/Visitors/SelectClauseVisitor.cs

```diff
@@ -16,14 +16,16 @@ public class SelectClauseVisitor : RelinqExpressionVisitor
 		private HashSet<Expression> _hqlNodes;
 		private readonly ParameterExpression _inputParameter;
 		private readonly VisitorParameters _parameters;
+		private readonly HashSet<Expression> _hqlCandidates;
 		private int _iColumn;
 		private List<HqlExpression> _hqlTreeNodes = new List<HqlExpression>();
 		private readonly HqlGeneratorExpressionVisitor _hqlVisitor;
 
-		public SelectClauseVisitor(System.Type inputType, VisitorParameters parameters)
+		public SelectClauseVisitor(System.Type inputType, VisitorParameters parameters, HashSet<Expression> hqlCandidates)
 		{
 			_inputParameter = Expression.Parameter(inputType, "input");
 			_parameters = parameters;
+			_hqlCandidates = hqlCandidates;
 			_hqlVisitor = new HqlGeneratorExpressionVisitor(_parameters);
 		}
```

Comment:
Need to keep previous signature for preventing a binary breaking change.

```suggestion
                // Since v5.6
                [Obsolete("Use overload providing expressions to be executed with HQL.")]
		public SelectClauseVisitor(System.Type inputType, VisitorParameters parameters) :
		    this(inputType, parameters, new HashSet<Expression>())
		{ }
		
		public SelectClauseVisitor(System.Type inputType, VisitorParameters parameters, HashSet<Expression> hqlCandidates)
		{
			_inputParameter = Expression.Parameter(inputType, "input");
			_parameters = parameters;
			_hqlCandidates = hqlCandidates;
			_hqlVisitor = new HqlGeneratorExpressionVisitor(_parameters);
		}
```